src/main/scala/cromwell/binding/WdlNamespace.scala

```diff
@@ -42,345 +128,186 @@ object WdlNamespace {
+      case Seq(x) => NamespaceWithWorkflow(ast, x, namespace, imports, namespaces, tasks, terminalMap, importedWorkflows, wdlSyntaxErrorFormatter)
+      case doh => throw new SyntaxError(wdlSyntaxErrorFormatter.tooManyWorkflows(doh.asJava))
     }
   }
 
+
   /**
-   * Confirm all required inputs are present and attempt to coerce raw inputs to `WdlValue`s.
-   * This can fail if required raw inputs are missing or if the values for a specified raw input
-   * cannot be coerced to the target type of the input as specified in the namespace.
+   * Given a name, a collection of WdlNamespaces and a collection of Tasks will attempt to find a Task
+   * with that name within the WdlNamespaces
    */
-  def coerceRawInputs(rawInputs: WorkflowRawInputs): Try[WorkflowCoercedInputs] = {
-
-    def coerceRawInput(fqn: FullyQualifiedName, wdlType: WdlType): Try[WdlValue] = fqn match {
-      case _ if rawInputs.contains(fqn) =>
-        val rawInput = rawInputs.get(fqn).get
-        wdlType.coerceRawValue(rawInput).recoverWith {
-          case e => Failure(new UnsatisfiedInputsException(s"Failed to coerce input $fqn value $rawInput of ${rawInput.getClass} to $wdlType."))
-        }
-      case _ => Failure(new UnsatisfiedInputsException(s"Required workflow input '$fqn' not specified."))
-    }
-
-    val tryCoercedValues = workflows.head.inputs.map { case (fqn, wdlType) =>
-      fqn -> coerceRawInput(fqn, wdlType)
-    }
-
-    val (successes, failures) = tryCoercedValues.partition { case (_, tryValue) => tryValue.isSuccess }
-    if (failures.isEmpty) {
-      Try(successes.map { case (key, tryValue) => key -> tryValue.get })
-    } else {
-      val message = failures.values.collect { case f: Failure[_] => f.exception.getMessage }.mkString("\n")
-      Failure(new UnsatisfiedInputsException(message))
-    }
+  def findTask(name: String, namespaces: Seq[WdlNamespace], tasks: Seq[Task]): Option[Task] = {
+    if (name.contains(".")) {
+      val parts = name.split("\\.", 2)
+      namespaces find {_.importedAs == Some(parts(0))} flatMap {x => findTask(parts(1), x.namespaces, x.tasks)}
+    } else tasks.find(_.name == name)
   }
 
+  private def localImportResolver(path: String): WdlSource = readFile(new File(path))
+  private def readFile(wdlFile: File): WdlSource = FileUtil.slurp(wdlFile)
+}
+
+object NamespaceWithWorkflow {
   /**
-   * Validate the following things about the AST:
-   *
+   * Validates:
    * 1) All `Call` blocks reference tasks that exist
    * 2) All `Call` inputs reference actual variables on the corresponding task
-   * 3) Tasks do not have duplicate inputs
-   * 4) Calls do not reference the same task input more than once
-   * 5) Tasks in this namespace have unique names
-   * 6) Tasks and namespaces don't have overlapping names
-   * 7) `Call` input expressions (right-hand side) should only use the MemberAccess
+   * 3) Calls do not reference the same task input more than once
```

Comment:
If I understand correctly I believe that's the check which 726 was complaining about.